tests/server/logs/test_stream.py

```diff
@@ -0,0 +1,579 @@
+    """Test distributor filters out logs that don't match"""
+    import asyncio
+    from unittest.mock import Mock
+
+    from prefect.server.logs.stream import distributor
+
+    # Create a mock message
+    mock_message = Mock()
+    mock_message.data = sample_log1.model_dump_json().encode()
+    mock_message.attributes = {"log_id": "test"}
+
+    # Create a filter that won't match (level too high)
+    queue = asyncio.Queue()
+    filter = LogFilter(level=LogFilterLevel(ge_=50))  # ERROR level or higher
+
+    # Add to global subscribers
+    from prefect.server.logs.stream import filters, subscribers
+
+    subscribers.add(queue)
+    filters[queue] = filter
+
+    try:
+        async with distributor() as handler:
+            await handler(mock_message)
+
+            # Queue should be empty because log was filtered out
+            assert queue.empty()
+    finally:
+        # Clean up
+        if queue in subscribers:
+            subscribers.remove(queue)
+        if queue in filters:
+            del filters[queue]
+
+
+@pytest.mark.asyncio
+async def test_distributor_message_handler_queue_full(sample_log1):
+    """Test distributor handles full queues gracefully"""
+    import asyncio
+    from unittest.mock import Mock
+
+    from prefect.server.logs.stream import distributor
+
+    # Create a mock message
+    mock_message = Mock()
+    mock_message.data = sample_log1.model_dump_json().encode()
+    mock_message.attributes = {"log_id": "test"}
+
+    # Create a queue and fill it
+    queue = asyncio.Queue(maxsize=1)
+    await queue.put("dummy")  # Fill the queue
+    filter = LogFilter()
+
+    # Add to global subscribers
+    from prefect.server.logs.stream import filters, subscribers
+
+    subscribers.add(queue)
+    filters[queue] = filter
+
+    try:
+        async with distributor() as handler:
+            # Should not raise an exception even with full queue
+            await handler(mock_message)
+    finally:
+        # Clean up
+        if queue in subscribers:
+            subscribers.remove(queue)
+        if queue in filters:
+            del filters[queue]
+
+
+@pytest.mark.asyncio
+async def test_start_stop_distributor():
+    """Test starting and stopping the distributor"""
+    from prefect.server.logs.stream import (
+        start_distributor,
+        stop_distributor,
+    )
+
+    # Ensure clean initial state
+    await stop_distributor()
+
+    try:
+        # Initially should be None
+        from prefect.server.logs.stream import _distributor_task
+
+        assert _distributor_task is None
+
+        # Start distributor
+        await start_distributor()
+
+        from prefect.server.logs.stream import _distributor_task
+
+        assert _distributor_task is not None
+        assert not _distributor_task.done()
+
+    finally:
+        # Stop distributor
+        await stop_distributor()
+
+        from prefect.server.logs.stream import _distributor_task
+
+        assert _distributor_task is None
```

Comment:
Again with the `try-finally` here, can we check on the events tests to see how it's done there?